utils/core/src/test/kotlin/UtilsTest.kt

```diff
@@ -75,6 +75,35 @@ class UtilsTest : WordSpec({
             filterVersionNames("0.3", names) should beEmpty()
         }
 
+        "find names separated by underscores that include a prefix / suffix" {
+            val names = listOf(
+                "REL_5_2_2",
+                "REL_5_2_1",
+                "REL_5_2_0",
+                "REL_5_1_0",
+                "REL_5_0_0",
+                "before_junit5_update",
+                "before_ooxml_3rd_edition",
+                "REL_4_1_2",
+                "REL_4_1_1",
+                "REL_4_1_0",
+                "REL_4_0_1",
+                "REL_4_0_0_FINAL",
+                "REL_3_17_FINAL",
+                "REL_3_17_BETA1",
+                "REL_3_16_FINAL",
+                "REL_3_16_BETA2",
+                "REL_3_15_BETA1",
+                "REL_3_16_BETA1",
+                "REL_3_15_FINAL",
+                "REL_3_15_BETA3"
+            )
+
+            filterVersionNames("4.0.0", names) shouldHaveSingleElement "REL_4_0_0_FINAL"
```

Comment:
For a version like 3.16 it would still fail to find a tag because of the multiple suffixes?

Reply:
No, it would actually succeed (as not both prefix and suffix are part of `ignorablePrefixSuffixPattern`). I've added a test to demonstrate that.